Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8306,6 +8306,27 @@ void WebPageProxy::didGetImageForFindMatch(ImageBufferParameters&& parameters, S
     m_findMatchesClient->didGetImageForMatchResult(this, image.ptr(), matchIndex);
 }
 
+void WebPageProxy::setTextIndicatorFromFrame(FrameIdentifier frameID, TextIndicatorData&& indicatorData, uint64_t lifetime)
+{
+    RefPtr frame = WebFrameProxy::webFrame(frameID);
+    if (!frame)
+        return;
+
+    RefPtr parentFrame = frame->parentFrame();
+    if (!parentFrame) {
+        setTextIndicator(WTFMove(indicatorData), lifetime);
+        return;
+    }
+
+    auto parentFrameID = parentFrame->frameID();
+    auto textBoundingRect = indicatorData.textBoundingRectInRootViewCoordinates;
+    sendToProcessContainingFrame(parentFrameID, Messages::WebPage::RemoteViewCoordinatesToRootView(frameID, textBoundingRect), [protectedThis = Ref { *this }, indicatorData = WTFMove(indicatorData), parentFrame = WTFMove(parentFrame), lifetime](FloatRect rect) mutable {
```

Comment:
This seems like it might modify behavior with site isolation off.  We probably don't want that.

Reply:
When site isolation is off we should never get here. The root frame will always be the main frame so we will early return above on `!parentFrame`.